src/js/grapl-cdk/lib/dgraph.ts

```diff
@@ -0,0 +1,195 @@
+import * as cdk from "@aws-cdk/core";
+import * as ec2 from "@aws-cdk/aws-ec2";
+import * as ecs from "@aws-cdk/aws-ecs";
+import * as servicediscovery from "@aws-cdk/aws-servicediscovery";
+
+class Zero extends cdk.Construct {
+    readonly name: string;
+
+    constructor(
+        scope: cdk.Construct,
+        graph_name: string,
+        id: string,
+        cluster: ecs.Cluster,
+        peer: string,
+        idx: number
+    ) {
+        super(scope, graph_name + id);
+
+        const zeroTask = new ecs.Ec2TaskDefinition(
+            this, 
+            id,
+            {
+                networkMode: ecs.NetworkMode.AWS_VPC,
+            }
+        );
+
+        const command = ["dgraph", "zero", `--my=${id}.${graph_name}.grapl:5080`,
+            "--replicas=3",
+            `--idx=${idx}`,
+            "--alsologtostderr"];
+
+        if (peer) {
+            command.push(`--peer=${peer}.${graph_name}.grapl:5080`);
+        }
+
+        const logDriver = new ecs.AwsLogDriver({
+            streamPrefix: `ecs${graph_name + id}`,
+        });
+
+        zeroTask.addContainer(id + 'Container', {
+            // --my is our own hostname (graph + id)
+            // --peer is the other dgraph zero hostname
+            image: ecs.ContainerImage.fromRegistry("dgraph/dgraph"),
+            command,
+            logging: logDriver,
+            memoryReservationMiB: 1024,
+        });
+
+        const zeroService = new ecs.Ec2Service(this, id + 'Service', {
+            cluster,  // Required
+            taskDefinition: zeroTask,
+            cloudMapOptions: {
+                name: id,
+                dnsRecordType: servicediscovery.DnsRecordType.A,
+                dnsTtl: cdk.Duration.seconds(300),
+            }
+        });
+
+        this.name = `${id}.${graph_name}.grapl`;
+
+        zeroService.connections.allowFromAnyIpv4(
+            ec2.Port.allTcp()
+        );
+    }
+}
+
+class Alpha extends cdk.Construct {
+    readonly name: string;
+
+    constructor(
+        scope: cdk.Construct,
+        graph_name: string,
+        id: string,
+        cluster: ecs.Cluster,
+        zero: string
+    ) {
+        super(scope, graph_name + id);
+
+        const alphaTask = new ecs.Ec2TaskDefinition(
+            this,
+            id,
+            {
+                networkMode: ecs.NetworkMode.AWS_VPC,
+            }
+        );
+
+        const logDriver = new ecs.AwsLogDriver({
+            streamPrefix: `ecs${graph_name + id}`,
+        });
+
+        alphaTask.addContainer(id + graph_name + 'Container', {
+            image: ecs.ContainerImage.fromRegistry("dgraph/dgraph"),
+            command: [
+                "dgraph", "alpha", `--my=${id}.${graph_name}.grapl:7080`,
+                "--lru_mb=1024", `--zero=${zero}.${graph_name}.grapl:5080`,
```

Comment:
If possible we should calculate the lru_mb.

DGraph represents 30% of system RAM, so we should do 30% of whatever we reserve for this container.

Reply:
I think we should come back to this one. We're launching a t3a.large, which has 32GB RAM, but default to one instance reserving soft limit of 2GB. I'll make a task.